### ISSUE TITLE
Escaped quotes from XProperty values are decoded incorrectly generating too many back-slashes

### ISSUE THREAD

**Issue by danielgrigore** [ical4j/ical4j#513]
**Escaped quotes from XProperty values are decoded incorrectly generating too many back-slashes**

When having a custom property that contains quotes they are encoded correctly but when parsing again that ics content is not decoded as it should be generating too many backslashes. If the operation is repeated one might end up with very large ics files. 

**Steps to reproduce the behavior**
1. Create an event
2. Create a custom property (`XProperty`) that contains quotes
3. Export the event to ics format
4. Use `CalendarBuilder` to parse the generated content
5. Compare original event with the parsed one
6. Optional repeat step 4 and 5 a couple of times 

Here's the code:
```
Calendar original = new Calendar();
VEvent vEvent = new VEvent();
original.getComponents().add(vEvent);
XProperty msProp = new XProperty("X-MICROSOFT-LOCATIONS", "[{\"DisplayName\":\"Microsoft Teams Meeting\", \"LocationCode\":\"013454\"}]");
vEvent.getProperties().add(msProp);       

System.out.println(original.toString());
Calendar build = new CalendarBuilder().build(new ByteArrayInputStream(original.toString().getBytes()));
System.out.println(build);

assertEquals(original, build);
```
And here's how the generated ics will look like:
Original:
```
BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20210723T124607Z
X-MICROSOFT-LOCATIONS:[{\"DisplayName\":\"Microsoft Teams Meeting\"\, \"LocationCode\":\"013454\"}]
END:VEVENT
END:VCALENDAR
```
Parsed once:
```
BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20210723T124607Z
X-MICROSOFT-LOCATIONS:[{\\\"DisplayName\\\":\\\"Microsoft Teams Meeting\\\"\\\, \\\"LocationCode\\\":\\\"013454\\\"}]
END:VEVENT
END:VCALENDAR
```
Parsed twice:
```
BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20210723T124607Z
X-MICROSOFT-LOCATIONS:[{\\\\\\\"DisplayName\\\\\\\":\\\\\\\"Microsoft Teams Meeting\\\\\\\"\\\\\\\, \\\\\\\"LocationCode\\\\\\\":\\\\\\\"013454\\\\\\\"}]
END:VEVENT
END:VCALENDAR
```

**Expected behavior**
The parsed ics should be the same as the original calendar ics.

**Environment (please complete the following information):**

1.  Linux: Ubuntu: 20.04.02
2. Java Version: 8 (openjdk version "1.8.0_292")
3. ical4j Version: 3.0.27

**Additional context**
I observed this behavior on some older version Outlook generated events.
As far as I checked it looks like the issue is in `PropertyCodec` class but I could not identify a solution quickly.

Please let me know if you need any details.
Thanks,

**Comment by danielgrigore** [comment]
Hi @benfortuna ,
Additionally, is there a way of telling the parser not to read all/some custom properties.
Thanks.

**Comment by benfortuna** [comment]
@danielgrigore thanks for letting me know. I've added fix in develop branch to ensure XProperty values are correctly decoded. See #514

Also, I've added basic support for ignoring property names in the `PropertyBuilder` class, which may help to configure which properties you want to skip. Currently it will throw an `IllegalArgumentException` in `PropertyBuilder.build()` if configured to ignore a property (may require some further work to expose this functionality to the developer).

**Comment by benfortuna** [comment]
Included in release 3.0.29.

**Comment by danielgrigore** [comment]
Hi @benfortuna ,

Many thanks for the quick fix, I tested it in `3.0.29` version and it worked correctly.
I also played around a little bit with `ContentHandlerContext` and configured some properties to be ignored and that also worked.
Great job,
Cheers.
